examples/flutter_firebaseauth/lib/authentication/authentication_state.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:equatable/equatable.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+
+abstract class AuthenticationState extends Equatable {}
+
+class AuthenticationUninitialized extends AuthenticationState {
+  @override
+  String toString() => 'AuthenticationUninitialized';
+}
+
+class AuthenticationAuthenticated extends AuthenticationState {
+  FirebaseUser user;
+
+  AuthenticationAuthenticated({this.user});
```

Comment:
If you're going to extend Equatable, you need to pass the props to super:

```dart
AuthenticationAuthenticated({this.user}) : super([user]);
```

Reply:
Sorry didn't know how to use Equatable(another copy-paste overlook).